packages/sync/src/utils/saveIncomingChanges.ts

```diff
@@ -26,47 +34,50 @@ export const saveChangesForModel = async (
   const idToExistingRecord: Record<number, (typeof existingRecords)[0]> = Object.fromEntries(
     existingRecords.map((e: any) => [e.id, e]),
   );
-  // follow the same pattern for incoming records
-  // https://github.com/beyondessential/tamanu/pull/4854#discussion_r1403828225
-  const idToIncomingRecord: { [key: number]: (typeof changes)[0] } = Object.fromEntries(
-    changes.filter(c => c.data.id).map(e => [e.data.id, e]),
-  );
-
-  const idsForUpdate = new Set(existingRecords.map((existing: any) => existing.id));
-
-  const recordsForCreate = changes
-    .filter(c => idToExistingRecord[c.data.id] === undefined)
-    .map(({ data }) => {
-      return sanitizeData(data);
-    });
-  const recordsForUpdate = changes
-    .filter(r => idsForUpdate.has(r.data.id))
-    .map(({ data }) => {
-      return sanitizeData(data);
-    });
+  const recordsForCreate = [];
+  const recordsForUpdate = [];
+  const recordsForDelete = [];
+
+  for (const change of changes) {
+    const { data, isDeleted } = change;
+
+    if (idToExistingRecord[data.id] === undefined) {
+      recordsForCreate.push(sanitizeData(data));
```

Comment:
What if `isDeleted` is also true - we don't want to create it in that case right?